src/Templates/ArchiveProductTemplatesCompatibility.php

```diff
@@ -93,24 +97,13 @@ function( $hook ) use ( $block_name ) {
 			return $content;
 		}
 
-		/**
-		 * The core/post-template has two different block names:
-		 * - core/post-template when the wrapper is rendered.
-		 * - core/null when the loop item is rendered.
-		 */
-		if (
-			'core/null' === $block_name &&
-			isset( $block['attrs']['__woocommerceNamespace'] ) &&
-			'woocommerce/product-query/product-template' === $block['attrs']['__woocommerceNamespace']
-		) {
-			$block_name = self::LOOP_ITEM_ID;
-		}
-
-		$supported_blocks = array_map(
-			function( $hook ) {
-				return $hook['block_name'];
-			},
-			array_values( $this->hook_data )
+		$supported_blocks = array_merge(
+			...array_map(
```

Comment:
The spread operator is available on PHP 7.4 ([source](https://stackoverflow.com/a/56072765)). Currently, the minium version of PHP supported by WooCommerce is PHP 7.3 https://developer.woocommerce.com/2023/06/05/new-requirement-for-woocommerce-8-2-php-7-4/

I think that we should revisit this line.

Reply:
AFAIK in this context it will work, as spreading array as an argument to another function was introduced in PHP 5.6 ([StackOverflow](https://stackoverflow.com/a/46861938), [PHP docs](https://www.php.net/manual/en/migration56.new-features.php) - look for "splat operator").

What was introduced in PHP 7.4 was a spread operator in array expressions ([docs](https://wiki.php.net/rfc/spread_operator_for_array)) that allows spreading one array into another one, like:

```
$arr = [3, 4];
$numbers = [1, 2, ...$arr, 5, 6]; // -> [1, 2, 3, 4, 5, 6]
```

From the docs:
> PHP has already supported [argument unpacking](https://wiki.php.net/rfc/argument_unpacking) (AKA spread operator) since 5.6. This RFC proposes to bring this feature to array expression.

---

However, I'll add an empty array as an argument to `array_merge` before spreading another array to make it safer in case we spread an empty array.

Reply:
TIL! Thanks for making the code safer 👍